dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ContentletIndexAPIImpl.java

```diff
@@ -659,34 +650,35 @@ public void addContentToIndex(final List<Contentlet> contentToIndex) {
     }
 
     private void indexContentListNow(final List<Contentlet> contentToIndex) {
-        final BulkRequestBuilder bulk = createBulkRequest(contentToIndex);
-        bulk.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-
-        putToIndex(bulk);
+        final BulkRequest bulkRequest = createBulkRequest(contentToIndex);
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        putToIndex(bulkRequest);
     } // indexContentListNow.
 
     private void indexContentListWaitFor(final List<Contentlet> contentToIndex) {
-
-        final BulkRequestBuilder bulk = createBulkRequest(contentToIndex);
-        bulk.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
-        putToIndex(bulk);
+        final BulkRequest bulkRequest = createBulkRequest(contentToIndex);
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
+        putToIndex(bulkRequest);
     } // indexContentListWaitFor.
 
     private void indexContentListDefer(final List<Contentlet> contentToIndex) {
-        final BulkRequestBuilder bulk = createBulkRequest(contentToIndex);
-        bulk.setRefreshPolicy(WriteRequest.RefreshPolicy.NONE);
-        putToIndex(bulk);
+        final BulkRequest bulkRequest = createBulkRequest(contentToIndex);
+        bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.NONE);
+        putToIndex(bulkRequest);
     } // indexContentListWaitFor.
 
     @Override
-    public void putToIndex(final BulkRequestBuilder bulk, ActionListener<BulkResponse> listener) {
-        if (bulk != null && bulk.numberOfActions() > 0) {
+    public void putToIndex(final BulkRequest bulkRequest,
+            final ActionListener<BulkResponse> listener) {
+        if (bulkRequest != null && bulkRequest.numberOfActions() > 0) {
+            bulkRequest.timeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS));
+
             if (listener != null) {
-                bulk.setTimeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS)).execute(listener);
+                RestHighLevelClientProvider.getInstance()
+                        .getClient().bulkAsync(bulkRequest, RequestOptions.DEFAULT, listener);
             } else {
-                BulkResponse response = bulk
-                        .setTimeout(TimeValue.timeValueMillis(INDEX_OPERATIONS_TIMEOUT_IN_MS))
-                        .execute().actionGet();
+                BulkResponse response = Sneaky.sneak(() -> RestHighLevelClientProvider.getInstance().getClient()
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)